src/plugin/adapter-registry/adapter-registry.ts

```diff
@@ -25,12 +25,21 @@ export class AdapterRegistry implements vscode.Disposable {
         context.subscriptions.push(this);
     }
 
-    registerAdapter(debugType: string, handlerToRegister: AdapterCapabilities): vscode.Disposable {
-        if (this.isDisposed) { return new vscode.Disposable(() => { }); }
-        this.handlers.set(debugType, handlerToRegister);
+    registerAdapter(debugTypes: string | string[], handlerToRegister: AdapterCapabilities): vscode.Disposable {
+        if (this.isDisposed) {
+            return new vscode.Disposable(() => { });
+        }
+
+        if (typeof debugTypes === 'string') {
+            debugTypes = [ debugTypes ];
+        }
+
+        for (const debugType of debugTypes) {
+            this.handlers.set(debugType, handlerToRegister);
+        }
+
         return new vscode.Disposable(() => {
-            const currentlyRegisteredHandler = this.handlers.get(debugType);
-            if (currentlyRegisteredHandler === handlerToRegister) {
```

Comment:
This code handles a corner case if different handlers are registered for the same type (in some distant future where there are many options and an API not under a single plugin's control). This code ensured that a later registration that overrode an earlier registration couldn't be revoked accidentally by the registrant of the first handler. Is there a good reason to remove that handling?

Reply:
OK, I'll add it back